### PR TITLE
Replace the project name "Chariott" in code and package names, with the component name "intent brokering" 

### LINKED ISSUE #149: [PBI] Evaluate removing "chariott" project name from package names, code etc.
## Description
We have received feedback that it might be better to remove the project name "Chariott" from package names, code, etc. and name everything for its functionality instead. Evaluate if we want to remove all references in code and make those changes.

### PR DESCRIPTION
**Pull request by ladatz**
Closes #149 

## Motivation and Context

Chariott is the project name. This PR removes "Chariott" as the naming of components, services, etc. from code. This helps to clarify that the Chariott project provides multiple components, including the Service Discovery service and the Intent Brokering service. These are two services which work independently from one another, and are not integrated in any way.

## Description

- Replace the name "chariott" with intent brokering, intent brokering service, and "intent", depending on the context, in all code, documentation, and examples.
- Note: diagrams have not been updated yet and will be in a later PR, as there is an issue with the workflow that needs to be addressed

### CODE REVIEW

File: intent_brokering/examples/applications/mock-vas/README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This code sample shows you how to use the mocked VAS for the dog mode scenario.
 The dog mode allows a car owner to keep their dog safe, while they are away from
 the car.
 
-## How-to consume a streaming service using Chariott
+## How-to consume a streaming service using Chariott intent brokering
```
Comment (jorchiu) [inline intent_brokering/examples/applications/mock-vas/README.md:7]:
```suggestion
## How-to consume a streaming service using Chariott Intent Brokering
```

File: intent_brokering/examples/applications/simulated-camera/README.md [hunk 2]
```diff
@@ -7,22 +7,22 @@ manual mode where you specify the frame rate event yourself.
 
 ## To run the application
 
-1. Start chariott runtime by executing `cargo run -p chariott` from the root directory
+1. Start intent brokering runtime by executing `cargo run -p intent_brokering` from the root directory
 2. Navigate to `intent_brokering/examples/applications/simulated-camera` directory
 3. Create a `images` directory and place there all the `.jpg` files you want the
```
Comment (jorchiu) [inline intent_brokering/examples/applications/simulated-camera/README.md:12]:
```suggestion
3. Create an `images` directory and place all the `.jpg` files there that you want the
```

File: intent_brokering/examples/applications/simulated-camera/README.md [hunk 3]
```diff
@@ -36,13 +36,13 @@ manual mode where you specify the frame rate event yourself.
    ```
 
    This makes a single request for a subscription intent for simulated camera
-   frames arriving at the desired frame per minute rate. Chariott then passes
+   frames arriving at the desired frame per minute rate. The broker then passes
```
Comment (jorchiu) [inline intent_brokering/examples/applications/simulated-camera/README.md:39]:
```suggestion
   frames arriving at the desired frame per minute rate. The Intent Broker then passes
```

**Review by ashbeitz — APPROVED** [review]
Approved

File: intent_brokering/examples/applications/mock-vas/README.md (hunk 1 above)
Comment (devkelley) [inline intent_brokering/examples/applications/mock-vas/README.md:7]:
```suggestion
## How-to consume a streaming service using Chariott Intent Broker
```

File: intent_brokering/examples/applications/mock-vas/README.md [hunk 4]
```diff
@@ -39,7 +39,7 @@ From the root directory:
    ```
 
    This makes a single request for 2 subscription intents for the
-   dog mode status and the temperature events to Chariott. Chariott then passes these
+   dog mode status and the temperature events to the intent broker. The intent broker then passes these
```
Comment (devkelley) [inline intent_brokering/examples/applications/mock-vas/README.md:42]:
```suggestion
   dog mode status and the temperature events to the Intent Broker. The Intent Broker then passes these
```

File: intent_brokering/proto/intent_brokering/provider/v1/provider.proto [hunk 5]
```diff
@@ -5,30 +5,30 @@
...
+package intent_brokering.provider.v1;
 
-import "chariott/common/v1/common.proto";
+import "intent_brokering/common/v1/common.proto";
 
 /**
 * The provider service definition.
 *
-* This service is used by chariott to communicate with the provider. Each service application
+* This service is used by intent_brokering to communicate with the provider. Each service application
```
Comment (devkelley) [inline intent_brokering/proto/intent_brokering/provider/v1/provider.proto:21]:
```suggestion
* This service is used by the intent broker to communicate with the provider. Each service application
```

File: intent_brokering/tools/charc.md [hunk 6]
```diff
@@ -24,10 +24,10 @@ Chariott. `charc` stands for _Chariott console_.
 
     charc inspect <namespace> <query>
 
-Issues the _inspect intent_ to Chariott for fullfilment by some provider
+Issues the _inspect intent_ to the intent broker for fullfilment by some provider
```
Comment (devkelley) [inline intent_brokering/tools/charc.md:27]:
```suggestion
Issues the _inspect intent_ to the Intent Broker for fullfilment by some provider
```

File: intent_brokering/tools/charc.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # `charc`
 
 `charc` is a Bash script for simple and quick session-based interactions with
-Chariott. `charc` stands for _Chariott console_.
+Chariott intent brokering. `charc` stands for _Chariott intent console_.
```
Comment (devkelley) [inline intent_brokering/tools/charc.md:4]:
```suggestion
Chariott Intent Brokering. `charc` stands for _Chariott intent console_.
```

File: intent_brokering/tools/charc.md [hunk 8]
```diff
@@ -36,7 +36,7 @@ for the `<namespace>` argument:
 
     charc discover <namespace>
 
-Issues the _discover intent_ to Chariott for fullfilment by some provider
+Issues the _discover intent_ to the intent broker for fullfilment by some provider
```
Comment (devkelley) [inline intent_brokering/tools/charc.md:39]:
```suggestion
Issues the _discover intent_ to the Intent Broker for fullfilment by some provider
```

File: intent_brokering/tools/charc.md [hunk 9]
```diff
@@ -48,7 +48,7 @@ connections.
 
     charc read <namespace> <key>
 
-Issues the _read intent_ to Chariott for fullfilment by some provider
+Issues the _read intent_ to the intent broker for fullfilment by some provider
```
Comment (devkelley) [inline intent_brokering/tools/charc.md:51]:
```suggestion
Issues the _read intent_ to the Intent Broker for fullfilment by some provider
```

File: intent_brokering/tools/charc.md [hunk 10]
```diff
@@ -57,7 +57,7 @@ providing the capability, e.g.:
 
     charc write <namespace> <key> <value>
 
-Issues the _write intent_ to Chariott for fullfilment by some provider
+Issues the _write intent_ to the intent broker for fullfilment by some provider
```
Comment (devkelley) [inline intent_brokering/tools/charc.md:60]:
```suggestion
Issues the _write intent_ to the Intent Broker for fullfilment by some provider
```

File: intent_brokering/tools/charc.md [hunk 11]
```diff
@@ -66,7 +66,7 @@ providing the capability, e.g.:
 
     charc invoke <namespace> <command> [(<type> <arg>)...]
 
-Issues the _invoke intent_ to Chariott for fullfilment by some provider
+Issues the _invoke intent_ to the intent broker for fullfilment by some provider
```
Comment (devkelley) [inline intent_brokering/tools/charc.md:69]:
```suggestion
Issues the _invoke intent_ to the Intent Broker for fullfilment by some provider
```

File: intent_brokering/tools/charc.md [hunk 12]
```diff
@@ -113,7 +113,7 @@ eventually to close all opened streams.
 
     charc subscribe <namespace> <source>
 
-Issues the _subscribe intent_ to Chariott for fullfilment by some provider
+Issues the _subscribe intent_ to the intent broker for fullfilment by some provider
```
Comment (devkelley) [inline intent_brokering/tools/charc.md:116]:
```suggestion
Issues the _subscribe intent_ to the Intent Broker for fullfilment by some provider
```

File: intent_brokering/tools/charc.md [hunk 13]
```diff
@@ -148,7 +148,7 @@ for `<namespace>` to open a streaming channel.
     charc show (req|request|rsp|response)
 
 Shows the last gRPC request (`req`) sent or response (`rsp`) received by the
-last sub-command that communicated with Chariott.
+last sub-command that communicated with the intent broker.
```
Comment (devkelley) [inline intent_brokering/tools/charc.md:151]:
```suggestion
last sub-command that communicated with the Intent Broker.
```

File: Dockerfile.intent_brokering [hunk 14]
```diff
@@ -5,18 +5,19 @@
...
+# Dockerfile for building Eclipse Chariott Intent Brokering runtime container
 #
-# This Dockerfile utilizes a two step build process. It builds Chariott with
-# statically linked dependencies (using musl) for a x86_64 architecture.
+# This Dockerfile utilizes a two step build process. It builds Intent Brokering
+# service with statically linked dependencies (using musl) for a x86_64
+# architecture.
 
 # Chariott user id
 ARG CHARIOTT_UID=10001
```
[the remark this answers is not in the document]
Comment (ladatz) [reply Dockerfile.intent_brokering:15]: I have left this in on purpose, as it's what we use in service_discovery as well. I will leave it as-is for now, and if we decide we want to change it, I'll do it in a follow up PR where I change it in both places. Created an issue to track: #253

**Review by jorchiu — APPROVED** [review]

**Pull request merged**